cli/src/main.rs

```diff
@@ -1,8 +1,89 @@
+use db::PgPool;
+use tokio;
+use clap::{Parser, Subcommand};
+use db::sqlx;
+use std::fs;
+use std::io::Write;
+use uuid::Uuid; // Ensure you have the `uuid` crate in your Cargo.toml
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct KnowledgeArgs {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    Login,
+    Signup,
+    DisplayUsers,
+}
+
 #[tokio::main]
 async fn main() -> color_eyre::Result<()> {
-    let _db_pool = db::setup_db_pool().await?;
+    let args = KnowledgeArgs::parse();
+    let db_pool = db::setup_db_pool().await?;
+
+    match args.command {
+        Command::Login => {
+            println!("Login not implemented yet")
+        },
+        Command::Signup => {
+            sign_up(&db_pool).await?;
+        },
+        Command::DisplayUsers => {
+            display_users(&db_pool).await?;
+        }
+    }
+
+    Ok(())
+}
+
+async fn persist_auth_session(user_id: Uuid) -> color_eyre::Result<()> {
+    if fs::metadata("auth.txt").is_ok() {
+        return Err(color_eyre::eyre::eyre!("auth.txt already exists"));
+    }
+
+    let mut file = fs::File::create("auth.txt")?;
+    writeln!(file, "{}", user_id)?;
 
-    println!("Hello, world!");
+    Ok(())
+}
+
+async fn add_user(pool: &PgPool, username: &str) -> color_eyre::Result<()> {
+    let result = sqlx::query!(
+        "INSERT INTO Users (user_name) VALUES ($1) RETURNING user_id",
+        username
+    )
+    .fetch_one(pool)
+    .await?;
```

Comment:
What do you think about moving this to the `db` create maybe as a `create_user` method and then calling that here?

That way we can isolate all the DB interations to the DB crate!
One nice thing that I think will be easier if we do that, is switching out this direct DB interaction for an API eventually.

The CLI talking to the DB 'directly' is great for getting started today, but if we were to ship this CLI to users we would NOT want them to be able to write to the DB directly, cause that would be chaos lol!
Instead we'd have our server write to the Database, based on API requests from the cli clients.

And I think that transition will be easier if we can "just" remove the `db` crate from the CLI and fill in all the gaps with API calls lol